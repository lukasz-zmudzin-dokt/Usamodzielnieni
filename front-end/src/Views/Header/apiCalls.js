import proxy from "config/api";

export const logoutUser = async (token) => {
    let url =
        proxy.account + "logout/";
    const headers = {
        Authorization: "token " + token,
        "Content-Type": "application/json",
    };

    const response = await fetch(url, { method: "POST", headers });
    if (response.status === 200) {
        return true;
    } else {
        throw response.status;
    }
};