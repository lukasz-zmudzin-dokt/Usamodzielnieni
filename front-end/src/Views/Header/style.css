.font {
    font-family: "MonoType", serif;
    font-size: 16px;
}
.register-color {
    background-color: #42B4A8;
}
.login-color {
    background-color: #4F74C6;
}
.notification-color {
    background-color: #ff884d;
}
.white {
    color: white!important;
}
.dropdown-menu {
    border-style: none;
    margin: 0;
    padding: 0;
}
.hamburger-toggle {
    border-style: none;
    background-color: transparent;
}
.hamburger-toggle:focus {
    outline: none;
}
.hamburger-menu-text {
    color: #5073C5!important;
    vertical-align: middle;
}
.hamburger-menu-logo {
    width: 35px;
    margin-left: 15px;
}

@media(min-width: 992px) {
    .account-dropdown-button {
        background-color: #6ac8bf!important;
        max-width: 303px;
        padding-left: 12px;
    }
    .navbar-left-button {
        background-color: #A63A53;
        color: white!important;
        padding-left: 20px!important;
        padding-right: 20px!important;
        text-align: center;
    }
    .logo-button {
        background-color: #A63A53;
        padding-left: 20px!important;
        padding-right: 20px!important;
    }
    .navbar-right-button {
        color: white!important;
        padding-left: 20px!important;
        padding-right: 20px!important;
        margin-left: 20px;
        text-align: center;
    }
    .logo {
        width: 200px;
        padding-bottom: 3px;
    }
    .account-dropdown-button-first {
        margin-top: 5px;
    }
}
@media(max-width: 991px) {
    .account-dropdown-button {
        background-color: #6ac8bf!important;
        text-align: center;
        margin-top: 0;
    }
    .navbar-left-button {
        background-color: #A63A53;
        color: white!important;
        padding-left: 20px!important;
        padding-right: 20px!important;
        text-align: center;
    }
    .logo-button {
        background-color: #A63A53;
        padding-left: 20px!important;
        padding-right: 20px!important;
        text-align: center;
    }
    .navbar-right-button {
        color: white!important;
        text-align: center;
    }
    .logo {
        width: 200px;
        padding-bottom: 3px;
    }
}