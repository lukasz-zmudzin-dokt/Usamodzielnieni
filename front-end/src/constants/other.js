const DEFAULT_INPUT = "-- Wybierz --";

export { DEFAULT_INPUT };
