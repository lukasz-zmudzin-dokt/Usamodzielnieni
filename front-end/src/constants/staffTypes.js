export const staffTypes = {
  VERIFICATION: "staff_verification",
  CV: "staff_cv",
  JOBS: "staff_jobs",
  BLOG_CREATOR: "staff_blog_creator",
  BLOG_MODERATOR: "staff_blog_moderator",
};
