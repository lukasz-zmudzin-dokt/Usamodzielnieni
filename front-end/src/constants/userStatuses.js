export const userStatuses = {
  VERIFIED: "Verified",
  REJECTED: "Not verified",
  AWAITING: "Waiting for verification",
};
