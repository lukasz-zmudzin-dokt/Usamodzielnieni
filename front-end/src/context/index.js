export { UserContext, UserProvider } from "context/UserContext";
export { AlertContext, AlertProvider } from "context/AlertContext";
export { ChatContext, ChatProvider } from "context/ChatContext";
export {
  NotificationsContext,
  NotificationsProvider,
} from "context/NotificationsContext";
