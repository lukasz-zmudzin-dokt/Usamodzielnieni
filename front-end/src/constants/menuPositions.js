import { userTypes } from "./userTypes";
import { staffTypes } from "./staffTypes";
import { paths } from "./paths";

export default [       
    {
        name: "Kreator CV",
        path: paths.CV_CREATOR,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Moje CV",
        path: paths.MY_CVS,
        allowed: [userTypes.STANDARD],
        verified: true
    },
    {
        name: "Akceptacja CV",
        path: paths.CV_APPROVAL,
        allowed: [staffTypes.CV],
        verified: true
    },
    {
        name: "Oferty pracy",
        path: paths.JOB_OFFERS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj ofertę",
        path: paths.OFFER_FORM,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Moje oferty",
        path: paths.MY_OFFERS,
        allowed: [userTypes.EMPLOYER],
        verified: true
    },
    {
        name: "Blogi",
        path: paths.BLOG_PAGE,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER]
    },
    {
        name: "Dodaj post",
        path: paths.BLOG_FORM,
        allowed: [staffTypes.BLOG_CREATOR],
        verified: true
    },
    {
        name: "Telefony",
        path: paths.CONTACT_PAGE,
        allowed: undefined, //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Kroki",
        path: paths.STEPS,
        allowed: undefined //wszyscy są upoważnieni - zalogowani i niezalogowani
    },
    {
        name: "Chat",
        path: paths.CHATS,
        allowed: [userTypes.STANDARD, userTypes.STAFF, userTypes.EMPLOYER],
        verified: true,
    }
];
