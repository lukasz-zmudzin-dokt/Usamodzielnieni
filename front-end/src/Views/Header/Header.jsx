import React, { useContext, useState } from "react";
import { Nav, Navbar, NavDropdown } from "react-bootstrap";
import { IndexLinkContainer } from "react-router-bootstrap";
import { UserContext } from "context";
import { logoutUser } from "./apiCalls";
import "./style.css";
import { Redirect, useLocation } from "react-router-dom";
import logo from "assets/logo-white.png";
import menu from "assets/hamburger-menu-icon.svg";
import Alert from "react-bootstrap/Alert";
import Notifications from "./components/Notifications";
import { paths } from "constants/paths";
import { staffTypes } from "constants/staffTypes";

const Header = () => {
  const context = useContext(UserContext);
  const [isOpen, setIsOpen] = useState(false);
  const [logout, setLogout] = useState(false);
  const [error, setError] = useState(false);

  const handleOpen = () => {
    setIsOpen(true);
  };
  const handleClose = () => {
    setIsOpen(false);
  };

  const handleLogout = async () => {
    try {
      await logoutUser(context.token);
      setLogout(true);
      context.logout();
    } catch (e) {
      console.log(e);
      setError(true);
    }
  };
  const leftNav = (
    <Nav className="mr-auto">
      <IndexLinkContainer to={paths.DASHBOARD}>
        <Nav.Link className="logo-button">
          <img src={logo} className="logo" alt={"logo"} />
        </Nav.Link>
      </IndexLinkContainer>
      <IndexLinkContainer to={paths.CONTACT_PAGE}>
        <Nav.Link className="navbar-left-button ">KONTAKT</Nav.Link>
      </IndexLinkContainer>
      <IndexLinkContainer to={paths.BLOG_PAGE}>
        <Nav.Link className="navbar-left-button ">BLOG</Nav.Link>
      </IndexLinkContainer>
      <IndexLinkContainer to={paths.JOB_OFFERS}>
        <Nav.Link className="navbar-left-button ">OFERTY PRACY</Nav.Link>
      </IndexLinkContainer>
    </Nav>
  );

  const staffNavs = [
    {
      id: 0,
      link: paths.REGISTER_ADMIN,
      name: "ZAREJESTRUJ ADMINISTRATORA",
      group_type: staffTypes.VERIFICATION,
    },
    {
      id: 1,
      link: paths.USER_APPROVAL,
      name: "AKCEPTUJ NOWYCH UŻYTKOWNIKÓW",
      group_type: staffTypes.VERIFICATION,
    },
    {
      id: 2,
      link: paths.CV_APPROVAL,
      name: "AKCEPTACJA CV",
      group_type: staffTypes.CV,
    },
    {
      id: 3,
      link: paths.OFFER_APPROVAL,
      name: "AKCEPTUJ OFERTY PRACY",
      group_type: staffTypes.JOBS,
    },
    {
      id: 4,
      link: paths.CHATS,
      name: "CZAT",
      group_type: staffTypes.CHAT,
    },
  ];

  const accountDropdownNav =
    context.type === "standard" ? (
      <IndexLinkContainer to={paths.MY_CVS}>
        <NavDropdown.Item className="account-dropdown-button white">
          MOJE CV
        </NavDropdown.Item>
      </IndexLinkContainer>
    ) : context.type === "employer" ? (
      <IndexLinkContainer to={paths.MY_OFFERS}>
        <NavDropdown.Item className="account-dropdown-button white">
          MOJE OFERTY
        </NavDropdown.Item>
      </IndexLinkContainer>
    ) : context.type === "staff" && context.data !== undefined ? (
      <>
        <IndexLinkContainer to={paths.USER_LIST}>
          <NavDropdown.Item className="account-dropdown-button white">
            LISTA UŻYTKOWNIKÓW
          </NavDropdown.Item>
        </IndexLinkContainer>
        {staffNavs.map((nav) => {
          if (context.data.group_type.includes(nav.group_type)) {
            return (
              <IndexLinkContainer to={nav.link} key={nav.id}>
                <NavDropdown.Item className="account-dropdown-button white">
                  {nav.name}
                </NavDropdown.Item>
              </IndexLinkContainer>
            );
          } else {
            return null;
          }
        })}
      </>
    ) : null;

  const rightNav =
    context.token === undefined ? (
      <Nav>
        <IndexLinkContainer to={paths.REGISTER}>
          <Nav.Link className="navbar-right-button register-color">
            REJESTRACJA
          </Nav.Link>
        </IndexLinkContainer>
        <IndexLinkContainer to={paths.LOGIN}>
          <Nav.Link className="navbar-right-button login-color">
            LOGOWANIE
          </Nav.Link>
        </IndexLinkContainer>
      </Nav>
    ) : (
      <Nav>
        <Notifications location={useLocation} token={context.token} />
        <NavDropdown
          id={"myAccDropdown"}
          title={
            <span className="white" onClick={isOpen ? handleClose : handleOpen}>
              MOJE KONTO
            </span>
          }
          className="navbar-right-button register-color"
          onMouseEnter={handleOpen}
          onMouseLeave={handleClose}
          show={isOpen}
        >
          <IndexLinkContainer to={paths.USER}>
            <NavDropdown.Item className="account-dropdown-button account-dropdown-button-first white">
              MÓJ PROFIL
            </NavDropdown.Item>
          </IndexLinkContainer>
          {accountDropdownNav}
        </NavDropdown>
        <Nav.Link
          className="navbar-right-button login-color"
          onClick={handleLogout}
        >
          WYLOGUJ
        </Nav.Link>
      </Nav>
    );

  const errMsg = (
    <Alert variant="danger" className="w-100 mt-2">
      Wystąpił błąd.
    </Alert>
  );

  return (
    <Navbar
      collapseOnSelect
      expand="lg"
      sticky="top"
      className="font p-3 justify-content-end"
    >
      <Navbar.Toggle
        aria-controls="responsive-navbar-nav"
        bsPrefix="hamburger-toggle"
      >
        <p className="hamburger-menu-text">MENU</p>
        <img
          src={menu}
          id={"menu"}
          alt={"menu"}
          className="hamburger-menu-logo"
        />
      </Navbar.Toggle>
      <Navbar.Collapse id="responsive-navbar-nav">
        {leftNav}
        {rightNav}
      </Navbar.Collapse>
      {logout ? <Redirect to={paths.DASHBOARD} /> : null}
      {error ? errMsg : null}
    </Navbar>
  );
};

export default Header;
