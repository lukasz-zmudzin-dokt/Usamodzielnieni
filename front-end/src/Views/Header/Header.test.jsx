import React from "react";
import {fireEvent, render, waitForElement} from "@testing-library/react";
import {BrowserRouter, MemoryRouter} from "react-router-dom";
import {UserProvider} from "context";
import Header from "./Header";
import {userStatuses} from "constants/userStatuses";
import {staffTypes} from "constants/staffTypes";

describe('Header', () => {
    let fetchCheck;
    const user = {
        notLogged: {
            token: undefined,
            type: undefined,
            data: undefined,
            login: jest.fn(),
            logout: jest.fn()
        },
        standard : {
            token: "123",
            type: "standard",
            data: {
                first_name: "Jan",
                last_name: "Kowalski",
                status: userStatuses.VERIFIED,
                username: "123123",
            },
            login: jest.fn(),
            logout: jest.fn()
        },
        employer: {
            token: "123",
            type: "employer",
            data: {
                first_name: "Jan",
                last_name: "Kowalski",
                status: userStatuses.VERIFIED,
                username: "123123",
            },
            login: jest.fn(),
            logout: jest.fn()
        },
        staff: {
            token: "123",
            type: "staff",
            data: {
                first_name: "Jan",
                last_name: "Kowalski",
                username: "123123",
                group_type: [staffTypes.VERIFICATION, staffTypes.CV, staffTypes.JOBS]
            },
            login: jest.fn(),
            logout: jest.fn()
        }
    }

    beforeAll(() => {
        global.fetch = jest.fn().mockImplementation(() => {
            return new Promise((resolve, reject) => {
                switch (fetchCheck) {
                    case "fail":
                        resolve({ status: 500 });
                        break;
                    case "ok":
                        resolve({
                            status: 200,
                            json: () => Promise.resolve({}),
                        });
                        break;
                    default:
                        reject([]);
                }
            });
        });
    });

    it('should match snapshot - not logged in', async () => {
        const { container, getByText } = render (
            <UserProvider value={user.notLogged}>
                <MemoryRouter>
                    <BrowserRouter>
                        <Header />
                    </BrowserRouter>
                </MemoryRouter>
            </UserProvider>
        );
        await waitForElement(() => getByText("REJESTRACJA"));
        expect(container).toMatchSnapshot();
    });

    it('should match snapshot - logged in standard', async () => {
        const { container, getByText } = render (
            <UserProvider value={user.standard}>
                <MemoryRouter>
                    <BrowserRouter>
                        <Header />
                    </BrowserRouter>
                </MemoryRouter>
            </UserProvider>
        );
        await waitForElement(() => getByText("MOJE KONTO"));
        expect(container).toMatchSnapshot();
    });

    it('should match snapshot - logged in employer', async () => {
        const { container, getByText } = render (
            <UserProvider value={user.employer}>
                <MemoryRouter>
                    <BrowserRouter>
                        <Header />
                    </BrowserRouter>
                </MemoryRouter>
            </UserProvider>
        );
        await waitForElement(() => getByText("MOJE KONTO"));
        expect(container).toMatchSnapshot();
    });

    it('should match snapshot - logged in staff', async () => {
        const { container, getByText } = render (
            <UserProvider value={user.staff}>
                <MemoryRouter>
                    <BrowserRouter>
                        <Header />
                    </BrowserRouter>
                </MemoryRouter>
            </UserProvider>
        );
        await waitForElement(() => getByText("MOJE KONTO"));
        expect(container).toMatchSnapshot();
    });

    it('should logout user', async () => {
        fetchCheck = "ok";
        const { getByText } = render (
            <UserProvider value={user.standard}>
                <MemoryRouter>
                    <BrowserRouter>
                        <Header />
                    </BrowserRouter>
                </MemoryRouter>
            </UserProvider>
        );
        await waitForElement(() => getByText("WYLOGUJ"));
        fireEvent.click(getByText("WYLOGUJ"));
        expect(fetch).toHaveBeenCalled();
    });
    it('should logout user - error', async () => {
        fetchCheck = "fail";
        const { getByText } = render (
            <UserProvider value={user.standard}>
                <MemoryRouter>
                    <BrowserRouter>
                        <Header />
                    </BrowserRouter>
                </MemoryRouter>
            </UserProvider>
        );
        await waitForElement(() => getByText("WYLOGUJ"));
        fireEvent.click(getByText("WYLOGUJ"));
        expect(fetch).toHaveBeenCalled();
        await waitForElement(() => getByText("Wystąpił błąd."));
        expect(getByText("Wystąpił błąd.")).toBeInTheDocument();
    });
})