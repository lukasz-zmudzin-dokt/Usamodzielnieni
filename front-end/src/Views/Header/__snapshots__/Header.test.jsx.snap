// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot - logged in employer 1`] = `
<div>
  <nav
    class="font p-3 justify-content-end navbar navbar-expand-lg navbar-light sticky-top"
  >
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="hamburger-toggle collapsed"
      type="button"
    >
      <a
        class="hamburger-menu-text"
      >
        MENU
      </a>
      <img
        alt="menu"
        class="hamburger-menu-logo"
        id="menu"
        src="hamburger-menu-icon.svg"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="logo-button active nav-link"
          data-rb-event-key="/"
          href="/"
        >
          <img
            alt="logo"
            class="logo"
            src="logo-white.png"
          />
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/contact"
          href="/contact"
        >
          KONTAKT
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/blog"
          href="/blog"
        >
          BLOG
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/jobOffers"
          href="/jobOffers"
        >
          OFERTY PRACY
        </a>
      </div>
      <div
        class="navbar-nav"
      >
        <a
          class="navbar-right-button notification-color nav-link"
          href="#"
          role="button"
        >
          POWIADOMIENIA
        </a>
        <div
          class="navbar-right-button register-color dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle nav-link"
            href="#"
            role="button"
          >
            <span
              class="white"
            >
              MOJE KONTO
            </span>
          </a>
        </div>
        <a
          class="navbar-right-button login-color nav-link"
          href="#"
          role="button"
        >
          WYLOGUJ
        </a>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Header should match snapshot - logged in staff 1`] = `
<div>
  <nav
    class="font p-3 justify-content-end navbar navbar-expand-lg navbar-light sticky-top"
  >
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="hamburger-toggle collapsed"
      type="button"
    >
      <a
        class="hamburger-menu-text"
      >
        MENU
      </a>
      <img
        alt="menu"
        class="hamburger-menu-logo"
        id="menu"
        src="hamburger-menu-icon.svg"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="logo-button active nav-link"
          data-rb-event-key="/"
          href="/"
        >
          <img
            alt="logo"
            class="logo"
            src="logo-white.png"
          />
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/contact"
          href="/contact"
        >
          KONTAKT
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/blog"
          href="/blog"
        >
          BLOG
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/jobOffers"
          href="/jobOffers"
        >
          OFERTY PRACY
        </a>
      </div>
      <div
        class="navbar-nav"
      >
        <a
          class="navbar-right-button notification-color nav-link"
          href="#"
          role="button"
        >
          POWIADOMIENIA
        </a>
        <div
          class="navbar-right-button register-color dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle nav-link"
            href="#"
            role="button"
          >
            <span
              class="white"
            >
              MOJE KONTO
            </span>
          </a>
        </div>
        <a
          class="navbar-right-button login-color nav-link"
          href="#"
          role="button"
        >
          WYLOGUJ
        </a>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Header should match snapshot - logged in standard 1`] = `
<div>
  <nav
    class="font p-3 justify-content-end navbar navbar-expand-lg navbar-light sticky-top"
  >
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="hamburger-toggle collapsed"
      type="button"
    >
      <a
        class="hamburger-menu-text"
      >
        MENU
      </a>
      <img
        alt="menu"
        class="hamburger-menu-logo"
        id="menu"
        src="hamburger-menu-icon.svg"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="logo-button active nav-link"
          data-rb-event-key="/"
          href="/"
        >
          <img
            alt="logo"
            class="logo"
            src="logo-white.png"
          />
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/contact"
          href="/contact"
        >
          KONTAKT
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/blog"
          href="/blog"
        >
          BLOG
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/jobOffers"
          href="/jobOffers"
        >
          OFERTY PRACY
        </a>
      </div>
      <div
        class="navbar-nav"
      >
        <a
          class="navbar-right-button notification-color nav-link"
          href="#"
          role="button"
        >
          POWIADOMIENIA
        </a>
        <div
          class="navbar-right-button register-color dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle nav-link"
            href="#"
            role="button"
          >
            <span
              class="white"
            >
              MOJE KONTO
            </span>
          </a>
        </div>
        <a
          class="navbar-right-button login-color nav-link"
          href="#"
          role="button"
        >
          WYLOGUJ
        </a>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Header should match snapshot - not logged in 1`] = `
<div>
  <nav
    class="font p-3 justify-content-end navbar navbar-expand-lg navbar-light sticky-top"
  >
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="hamburger-toggle collapsed"
      type="button"
    >
      <a
        class="hamburger-menu-text"
      >
        MENU
      </a>
      <img
        alt="menu"
        class="hamburger-menu-logo"
        id="menu"
        src="hamburger-menu-icon.svg"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <a
          class="logo-button active nav-link"
          data-rb-event-key="/"
          href="/"
        >
          <img
            alt="logo"
            class="logo"
            src="logo-white.png"
          />
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/contact"
          href="/contact"
        >
          KONTAKT
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/blog"
          href="/blog"
        >
          BLOG
        </a>
        <a
          class="navbar-left-button nav-link"
          data-rb-event-key="/jobOffers"
          href="/jobOffers"
        >
          OFERTY PRACY
        </a>
      </div>
      <div
        class="navbar-nav"
      >
        <a
          class="navbar-right-button register-color nav-link"
          data-rb-event-key="/newAccount"
          href="/newAccount"
        >
          REJESTRACJA
        </a>
        <a
          class="navbar-right-button login-color nav-link"
          data-rb-event-key="/login"
          href="/login"
        >
          LOGOWANIE
        </a>
      </div>
    </div>
  </nav>
</div>
`;
